tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -119,26 +175,100 @@ Here, you see the simulated data. You can see two groups of cells that differ in
+ 2. optionally: an integer $k$ which is the dimension of the latent space (the space the original feature matrix is projected onto)
+    as input argument and defines the probabilistic model for p-PCA.
 
-Here, we run the inference with the NUTS sampler. Feel free to try [different samplers](https://turing.ml/stable/docs/library/#samplers).
+Specifically:
+
+ 1. it first extracts the dimension $D$ and number of instances $N$ of the input matrix;
+ 2. then draw $N$ samples of the latent variable $z$ with dimension $k$ from an MND;
+ 3. then draw samples of each entries of the projection matrix $w$ (with dimension $D \times k$), from a standard normal;
+ 4. then we assume uniform offset for each dimension $d \in D$, and draw samples from an MND for each dimension;
+ 5. Finally, we iterate through each dimension in $D$, and define an MND (sampling distribution, likelihood), assuming all instances in that dimension follow an MND with mean $w \times z + mu_d$, for all instances in that dimension.
+
+#### Step 4: Sampling-based inference of the pPCA model
+
+Here we aim to perform MCMC sampling to infer the projection matrix $w_{D \times k}$, the latent variable matrix $z_{k \times N}$, and the offsets $mu_{D \times 1}$.
+
+We run the inference using the NUTS sampler, of which the chain length is set to be 500. By default, the NUTS sampler samples 1 chain.
+You are free to try [different samplers](https://turing.ml/stable/docs/library/#samplers).
 
 ```julia
-ppca = pPCA(expression_matrix)
+k = 2 # k is the dimension of the projected space, i.e. the number of principal components/axes of choice
+ppca = pPCA(expression_matrix, k) # instantiate the probabilistic model
 chain_ppca = sample(ppca, NUTS(), 500);
 ```
 
-### pPCA control
+The sampling process takes about 7 mins. The samples are saved in the Chains struc `chain_ppca`, whose shape can be checked:
+
+```julia
+size(chain_ppca)
+```
+
+The Chains struc `chain_ppca` also contains the sampling info such as rhat, ess, mean estimates, etc. You can print it to check these quantities.
+
+#### Step 5: diagnostics and posterior predictive checks
+
+Diagnostics of the inference results, i.e. convergence (r-hat), effective samples size (ess), acceptance rate, etc, can be performed using the `MCMCChain` package.
+Here we plot the sample trajectories for 1 chain:
+
+```julia
+plot(chain_ppca[:, 1, :]; xlabel="iteration_no", ylabel="sample_value")
+```
+
+We observe that, the samples are not clustered around a certain value, which indicates equilibrium may not achieve. This could be associated with the probabilistic model and the sampling settings used.
+
+Posterior predictive check. We reconstruct the input data from our parameter estimates, using the posterior mean as parameter estimates.
+We first retrieve the samples for the projection matrix `w` from `chain_ppca`. This can be done using the Julia `group(chain, parameter name)` function.
```

Comment:
```suggestion
We first retrieve the samples for the projection matrix `w` from `chain_ppca`. This can be done using the following function: `group(chain, parameter name)`.
```